File: api/src/main/java/com/ruby/rubia_server/core/service/CustomerService.java
package com.ruby.rubia_server.core.service;

import com.ruby.rubia_server.core.dto.CreateCustomerDTO;
import com.ruby.rubia_server.core.dto.CustomerDTO;
import com.ruby.rubia_server.core.dto.UpdateCustomerDTO;
import com.ruby.rubia_server.core.entity.Customer;
import com.ruby.rubia_server.core.entity.Company;
import com.ruby.rubia_server.core.repository.CustomerRepository;
import com.ruby.rubia_server.core.repository.CompanyRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Service
@RequiredArgsConstructor
@Slf4j
@Transactional
public class CustomerService {
    
    private final CustomerRepository customerRepository;
    private final CompanyRepository companyRepository;
    
    public CustomerDTO create(CreateCustomerDTO createDTO, UUID companyId) {
        log.info("Creating customer with phone: {} for company: {}", createDTO.getPhone(), companyId);
        
        // Buscar a empresa
        Company company = companyRepository.findById(companyId)
                .orElseThrow(() -> new IllegalArgumentException("Empresa não encontrada"));
        
        // Normalizar telefone antes de verificar duplicação
        String normalizedPhone = normalizePhoneNumber(createDTO.getPhone());
        
        if (customerRepository.existsByPhoneAndCompanyId(normalizedPhone, companyId)) {
            throw new IllegalArgumentException("Cliente com telefone '" + normalizedPhone + "' já existe nesta empresa");
        }
        
        if (createDTO.getWhatsappId() != null && customerRepository.existsByWhatsappIdAndCompanyId(createDTO.getWhatsappId(), companyId)) {
            throw new IllegalArgumentException("Cliente com WhatsApp ID '" + createDTO.getWhatsappId() + "' já existe nesta empresa");
        }
        
        Customer customer = Customer.builder()
                .phone(normalizedPhone) // Usar o telefone normalizado
                .name(createDTO.getName())
                .whatsappId(createDTO.getWhatsappId())
                .profileUrl(createDTO.getProfileUrl())
                .isBlocked(createDTO.getIsBlocked() != null ? createDTO.getIsBlocked() : false)
                .sourceSystemName(createDTO.getSourceSystemName())
                .sourceSystemId(createDTO.getSourceSystemId())
                .importedAt(createDTO.getImportedAt())
                .birthDate(createDTO.getBirthDate())
                .lastDonationDate(createDTO.getLastDonationDate())
                .bloodType(createDTO.getBloodType())
                .height(createDTO.getHeight())
                .weight(createDTO.getWeight())
                .addressStreet(createDTO.getAddressStreet())
                .addressNumber(createDTO.getAddressNumber())
                .addressComplement(createDTO.getAddressComplement())
                .addressPostalCode(createDTO.getAddressPostalCode())
                .addressCity(createDTO.getAddressCity())
                .addressState(createDTO.getAddressState())
                .company(company)
                .build();
        
        customer.setNextEligibleDonationDate(createDTO.getNextEligibleDonationDate());
        
        Customer saved = customerRepository.save(customer);
        log.info("Customer created successfully with id: {}", saved.getId());
        
        return toDTO(saved);
    }
    
    @Transactional(readOnly = true)
    public CustomerDTO findById(UUID id, UUID companyId) {
        log.debug("Finding customer by id: {} for company: {}", id, companyId);
        
        Customer customer = customerRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Cliente não encontrado"));
        
        // Validate customer belongs to company
        if (!customer.getCompany().getId().equals(companyId)) {
            throw new IllegalArgumentException("Cliente não pertence a esta empresa");
        }
        
        return toDTO(customer);
    }
    
    public CustomerDTO update(UUID id, UpdateCustomerDTO updateDTO, UUID companyId) {
        log.info("Updating customer with id: {} for company: {}", id, companyId);
        
        Customer customer = customerRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Cliente não encontrado"));
        
        // Validate customer belongs to company
        if (!customer.getCompany().getId().equals(companyId)) {
            throw new IllegalArgumentException("Cliente não pertence a esta empresa");
        }
        
        if (updateDTO.getPhone() != null) {
            // Normalizar telefone antes de verificar duplicação
            String normalizedPhone = normalizePhoneNumber(updateDTO.getPhone());
            if (!normalizedPhone.equals(customer.getPhone())) {
                if (customerRepository.existsByPhoneAndCompanyId(normalizedPhone, companyId)) {
                    throw new IllegalArgumentException("Cliente com telefone '" + normalizedPhone + "' já existe nesta empresa");
                }
            }
            customer.setPhone(normalizedPhone);
        }
        
        if (updateDTO.getName() != null) {
            customer.setName(updateDTO.getName());
        }
        
        if (updateDTO.getWhatsappId() != null) {
            if (!updateDTO.getWhatsappId().equals(customer.getWhatsappId())) {
                if (customerRepository.existsByWhatsappIdAndCompanyId(updateDTO.getWhatsappId(), companyId)) {
                    throw new IllegalArgumentException("Cliente com WhatsApp ID '" + updateDTO.getWhatsappId() + "' já existe nesta empresa");
                }
            }
            customer.setWhatsappId(updateDTO.getWhatsappId());
        }
        
        if (updateDTO.getProfileUrl() != null) {
            customer.setProfileUrl(updateDTO.getProfileUrl());
        }
        
        if (updateDTO.getIsBlocked() != null) {
            customer.setIsBlocked(updateDTO.getIsBlocked());
        }

        Optional.ofNullable(updateDTO.getSourceSystemName()).ifPresent(customer::setSourceSystemName);
        Optional.ofNullable(updateDTO.getSourceSystemId()).ifPresent(customer::setSourceSystemId);
        Optional.ofNullable(updateDTO.getImportedAt()).ifPresent(customer::setImportedAt);
        Optional.ofNullable(updateDTO.getBirthDate()).ifPresent(customer::setBirthDate);
        Optional.ofNullable(updateDTO.getLastDonationDate()).ifPresent(customer::setLastDonationDate);
        Optional.ofNullable(updateDTO.getNextEligibleDonationDate()).ifPresent(customer::setNextEligibleDonationDate);
        Optional.ofNullable(updateDTO.getBloodType()).ifPresent(customer::setBloodType);
        Optional.ofNullable(updateDTO.getHeight()).ifPresent(customer::setHeight);
        Optional.ofNullable(updateDTO.getWeight()).ifPresent(customer::setWeight);
        Optional.ofNullable(updateDTO.getAddressStreet()).ifPresent(customer::setAddressStreet);
        Optional.ofNullable(updateDTO.getAddressNumber()).ifPresent(customer::setAddressNumber);
        Optional.ofNullable(updateDTO.getAddressComplement()).ifPresent(customer::setAddressComplement);
        Optional.ofNullable(updateDTO.getAddressPostalCode()).ifPresent(customer::setAddressPostalCode);
        Optional.ofNullable(updateDTO.getAddressCity()).ifPresent(customer::setAddressCity);
        Optional.ofNullable(updateDTO.getAddressState()).ifPresent(customer::setAddressState);
        
        Customer updated = customerRepository.save(customer);
        log.info("Customer updated successfully");
        
        return toDTO(updated);
    }
    
    public CustomerDTO blockCustomer(UUID id, UUID companyId) {
        log.info("Blocking customer with id: {} for company: {}", id, companyId);
        
        Customer customer = customerRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Cliente não encontrado"));
        
        // Validate customer belongs to company
        if (!customer.getCompany().getId().equals(companyId)) {
            throw new IllegalArgumentException("Cliente não pertence a esta empresa");
        }
        
        customer.setIsBlocked(true);
        Customer updated = customerRepository.save(customer);
        
        log.info("Customer blocked successfully");
        return toDTO(updated);
    }
    
    public CustomerDTO unblockCustomer(UUID id, UUID companyId) {
        log.info("Unblocking customer with id: {} for company: {}", id, companyId);
        
        Customer customer = customerRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Cliente não encontrado"));
        
        // Validate customer belongs to company
        if (!customer.getCompany().getId().equals(companyId)) {
            throw new IllegalArgumentException("Cliente não pertence a esta empresa");
        }
        
        customer.setIsBlocked(false);
        Customer updated = customerRepository.save(customer);
        
        log.info("Customer unblocked successfully");
        return toDTO(updated);
    }
    
    public void delete(UUID id, UUID companyId) {
        log.info("Deleting customer with id: {} for company: {}", id, companyId);
        
        Customer customer = customerRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Cliente não encontrado"));
        
        // Validate customer belongs to company
        if (!customer.getCompany().getId().equals(companyId)) {
            throw new IllegalArgumentException("Cliente não pertence a esta empresa");
        }
        
        customerRepository.deleteById(id);
        log.info("Customer deleted successfully");
    }
    
    @Transactional(readOnly = true)
    public String normalizePhoneNumber(String phone) {
        if (phone == null) {
            return null;
        }
        
        // Remove all non-digit characters
        String digitsOnly = phone.replaceAll("\\D", "");
        
        // Add +55 prefix if not present
        if (digitsOnly.length() == 10 || digitsOnly.length() == 11) {
            // Brazilian phone without country code
            return "+55" + digitsOnly;
        } else if (digitsOnly.length() == 12 && digitsOnly.startsWith("55")) {
            // Brazilian phone with country code but without +
            return "+" + digitsOnly;
        } else if (digitsOnly.length() == 13 && digitsOnly.startsWith("55")) {
            // Brazilian phone with country code and +
            return "+" + digitsOnly;
        }
        
        return phone; // Return as is if format is unexpected
    }
    
    // Company-scoped methods
    @Transactional(readOnly = true)
    public List<CustomerDTO> findAllByCompany(UUID companyId) {
        log.debug("Finding all customers for company: {}", companyId);
        
        return customerRepository.findByCompanyId(companyId)
                .stream()
                .map(this::toDTO)
                .toList();
    }
    
    @Transactional(readOnly = true)
    public CustomerDTO findByPhoneAndCompany(String phone, UUID companyId) {
        log.debug("Finding customer by phone: {} for company: {}", phone, companyId);
        
        Optional<Customer> customer = customerRepository.findByPhoneAndCompanyId(phone, companyId);
        
        return customer.map(this::toDTO).orElse(null);
    }
    
    @Transactional(readOnly = true)
    public List<CustomerDTO> findActiveByCompany(UUID companyId) {
        log.debug("Finding active customers for company: {}", companyId);
        
        return customerRepository.findByIsBlockedFalseAndCompanyId(companyId)
                .stream()
                .map(this::toDTO)
                .toList();
    }
    
    @Transactional(readOnly = true)
    public List<CustomerDTO> searchByNameOrPhoneAndCompany(String searchTerm, UUID companyId) {
        log.debug("Searching customers by term: {} for company: {}", searchTerm, companyId);
        
        return customerRepository.searchByNameOrPhoneAndCompany(searchTerm, companyId)
                .stream()
                .map(this::toDTO)
                .toList();
    }
    
    @Transactional(readOnly = true)
    public long countActiveByCompany(UUID companyId) {
        return customerRepository.countActiveCustomersByCompany(companyId);
    }
    
    public void deleteAllByCompany(UUID companyId) {
        log.info("Deleting all customers for company: {}", companyId);
        
        List<Customer> customers = customerRepository.findByCompanyId(companyId);
        customerRepository.deleteAll(customers);
        
        log.info("Deleted {} customers for company: {}", customers.size(), companyId);
    }

    private CustomerDTO toDTO(Customer customer) {
        return CustomerDTO.builder()
                .id(customer.getId())
                .companyId(customer.getCompany() != null ? customer.getCompany().getId() : null)
                .phone(customer.getPhone())
                .name(customer.getName())
                .whatsappId(customer.getWhatsappId())
                .profileUrl(customer.getProfileUrl())
                .isBlocked(customer.getIsBlocked())
                .sourceSystemName(customer.getSourceSystemName())
                .sourceSystemId(customer.getSourceSystemId())
                .importedAt(customer.getImportedAt())
                .birthDate(customer.getBirthDate())
                .lastDonationDate(customer.getLastDonationDate())
                .nextEligibleDonationDate(customer.getNextEligibleDonationDate())
                .bloodType(customer.getBloodType())
                .height(customer.getHeight())
                .weight(customer.getWeight())
                .addressStreet(customer.getAddressStreet())
                .addressNumber(customer.getAddressNumber())
                .addressComplement(customer.getAddressComplement())
                .addressPostalCode(customer.getAddressPostalCode())
                .addressCity(customer.getAddressCity())
                .addressState(customer.getAddressState())
                .createdAt(customer.getCreatedAt())
                .updatedAt(customer.getUpdatedAt())
                .build();
    }
}

File: api-dotnet/Rubia.Server/Services/CustomerService.cs
using Microsoft.EntityFrameworkCore;
using Rubia.Server.Data;
using Rubia.Server.DTOs;
using Rubia.Server.Entities;
using Rubia.Server.Services.Interfaces;

namespace Rubia.Server.Services;

public class CustomerService : ICustomerService
{
    private readonly RubiaDbContext _context;
    private readonly IPhoneService _phoneService;
    private readonly ILogger<CustomerService> _logger;

    public CustomerService(RubiaDbContext context, IPhoneService phoneService, ILogger<CustomerService> logger)
    {
        _context = context;
        _phoneService = phoneService;
        _logger = logger;
    }

    public async Task<CustomerDto> CreateAsync(CreateCustomerDto createDto, Guid companyId)
    {
        _logger.LogInformation("Creating customer with phone: {Phone} for company: {CompanyId}", createDto.Phone, companyId);

        // Buscar a empresa
        var company = await _context.Companies.FirstOrDefaultAsync(c => c.Id == companyId);
        if (company == null)
        {
            throw new ArgumentException("Empresa não encontrada");
        }

        // Normalizar telefone antes de verificar duplicação
        var normalizedPhone = _phoneService.Normalize(createDto.Phone);

        if (await _context.Customers.AnyAsync(c => c.Phone == normalizedPhone && c.CompanyId == companyId))
        {
            throw new ArgumentException($"Cliente com telefone '{normalizedPhone}' já existe nesta empresa");
        }

        if (!string.IsNullOrEmpty(createDto.WhatsappId) && 
            await _context.Customers.AnyAsync(c => c.WhatsappId == createDto.WhatsappId && c.CompanyId == companyId))
        {
            throw new ArgumentException($"Cliente com WhatsApp ID '{createDto.WhatsappId}' já existe nesta empresa");
        }

        var customer = new Customer
        {
            Phone = normalizedPhone,
            Name = createDto.Name,
            WhatsappId = createDto.WhatsappId,
            ProfileUrl = createDto.ProfileUrl,
            IsBlocked = createDto.IsBlocked,
            SourceSystemName = createDto.SourceSystemName,
            SourceSystemId = createDto.SourceSystemId,
            ImportedAt = createDto.ImportedAt,
            BirthDate = createDto.BirthDate,
            LastDonationDate = createDto.LastDonationDate,
            NextEligibleDonationDate = createDto.NextEligibleDonationDate,
            BloodType = createDto.BloodType,
            Height = createDto.Height,
            Weight = createDto.Weight,
            AddressStreet = createDto.AddressStreet,
            AddressNumber = createDto.AddressNumber,
            AddressComplement = createDto.AddressComplement,
            AddressPostalCode = createDto.AddressPostalCode,
            AddressCity = createDto.AddressCity,
            AddressState = createDto.AddressState,
            Email = createDto.Email,
            Cpf = createDto.Cpf,
            Rg = createDto.Rg,
            RhFactor = createDto.RhFactor,
            CompanyId = companyId
        };

        _context.Customers.Add(customer);
        await _context.SaveChangesAsync();

        _logger.LogInformation("Customer created successfully with id: {Id}", customer.Id);
        return ToDto(customer);
    }

    public async Task<CustomerDto> FindByIdAsync(Guid id, Guid companyId)
    {
        _logger.LogDebug("Finding customer by id: {Id} for company: {CompanyId}", id, companyId);

        var customer = await _context.Customers
            .FirstOrDefaultAsync(c => c.Id == id);

        if (customer == null)
        {
            throw new ArgumentException("Cliente não encontrado");
        }

        // Validate customer belongs to company
        if (customer.CompanyId != companyId)
        {
            throw new ArgumentException("Cliente não pertence a esta empresa");
        }

        return ToDto(customer);
    }

    public async Task<CustomerDto> UpdateAsync(Guid id, UpdateCustomerDto updateDto, Guid companyId)
    {
        _logger.LogInformation("Updating customer with id: {Id} for company: {CompanyId}", id, companyId);

        var customer = await _context.Customers
            .FirstOrDefaultAsync(c => c.Id == id);

        if (customer == null)
        {
            throw new ArgumentException("Cliente não encontrado");
        }

        // Validate customer belongs to company
        if (customer.CompanyId != companyId)
        {
            throw new ArgumentException("Cliente não pertence a esta empresa");
        }

        if (updateDto.Phone != null)
        {
            // Normalizar telefone antes de verificar duplicação
            var normalizedPhone = _phoneService.Normalize(updateDto.Phone);
            if (normalizedPhone != customer.Phone)
            {
                if (await _context.Customers.AnyAsync(c => c.Phone == normalizedPhone && c.CompanyId == companyId))
                {
                    throw new ArgumentException($"Cliente com telefone '{normalizedPhone}' já existe nesta empresa");
                }
            }
            customer.Phone = normalizedPhone;
        }

        if (updateDto.Name != null) customer.Name = updateDto.Name;

        if (updateDto.WhatsappId != null)
        {
            if (updateDto.WhatsappId != customer.WhatsappId)
            {
                if (await _context.Customers.AnyAsync(c => c.WhatsappId == updateDto.WhatsappId && c.CompanyId == companyId))
                {
                    throw new ArgumentException($"Cliente com WhatsApp ID '{updateDto.WhatsappId}' já existe nesta empresa");
                }
            }
            customer.WhatsappId = updateDto.WhatsappId;
        }

        if (updateDto.ProfileUrl != null) customer.ProfileUrl = updateDto.ProfileUrl;
        if (updateDto.IsBlocked.HasValue) customer.IsBlocked = updateDto.IsBlocked.Value;
        if (updateDto.SourceSystemName != null) customer.SourceSystemName = updateDto.SourceSystemName;
        if (updateDto.SourceSystemId != null) customer.SourceSystemId = updateDto.SourceSystemId;
        if (updateDto.ImportedAt.HasValue) customer.ImportedAt = updateDto.ImportedAt;
        if (updateDto.BirthDate.HasValue) customer.BirthDate = updateDto.BirthDate;
        if (updateDto.LastDonationDate.HasValue) customer.LastDonationDate = updateDto.LastDonationDate;
        if (updateDto.NextEligibleDonationDate.HasValue) customer.NextEligibleDonationDate = updateDto.NextEligibleDonationDate;
        if (updateDto.BloodType != null) customer.BloodType = updateDto.BloodType;
        if (updateDto.Height.HasValue) customer.Height = updateDto.Height;
        if (updateDto.Weight.HasValue) customer.Weight = updateDto.Weight;
        if (updateDto.AddressStreet != null) customer.AddressStreet = updateDto.AddressStreet;
        if (updateDto.AddressNumber != null) customer.AddressNumber = updateDto.AddressNumber;
        if (updateDto.AddressComplement != null) customer.AddressComplement = updateDto.AddressComplement;
        if (updateDto.AddressPostalCode != null) customer.AddressPostalCode = updateDto.AddressPostalCode;
        if (updateDto.AddressCity != null) customer.AddressCity = updateDto.AddressCity;
        if (updateDto.AddressState != null) customer.AddressState = updateDto.AddressState;
        if (updateDto.Email != null) customer.Email = updateDto.Email;
        if (updateDto.Cpf != null) customer.Cpf = updateDto.Cpf;
        if (updateDto.Rg != null) customer.Rg = updateDto.Rg;
        if (updateDto.RhFactor != null) customer.RhFactor = updateDto.RhFactor;

        await _context.SaveChangesAsync();
        _logger.LogInformation("Customer updated successfully");

        return ToDto(customer);
    }

    public async Task<CustomerDto> BlockCustomerAsync(Guid id, Guid companyId)
    {
        _logger.LogInformation("Blocking customer with id: {Id} for company: {CompanyId}", id, companyId);

        var customer = await _context.Customers
            .FirstOrDefaultAsync(c => c.Id == id);

        if (customer == null)
        {
            throw new ArgumentException("Cliente não encontrado");
        }

        // Validate customer belongs to company
        if (customer.CompanyId != companyId)
        {
            throw new ArgumentException("Cliente não pertence a esta empresa");
        }

        customer.IsBlocked = true;
        await _context.SaveChangesAsync();

        _logger.LogInformation("Customer blocked successfully");
        return ToDto(customer);
    }

    public async Task<CustomerDto> UnblockCustomerAsync(Guid id, Guid companyId)
    {
        _logger.LogInformation("Unblocking customer with id: {Id} for company: {CompanyId}", id, companyId);

        var customer = await _context.Customers
            .FirstOrDefaultAsync(c => c.Id == id);

        if (customer == null)
        {
            throw new ArgumentException("Cliente não encontrado");
        }

        // Validate customer belongs to company
        if (customer.CompanyId != companyId)
        {
            throw new ArgumentException("Cliente não pertence a esta empresa");
        }

        customer.IsBlocked = false;
        await _context.SaveChangesAsync();

        _logger.LogInformation("Customer unblocked successfully");
        return ToDto(customer);
    }

    public async Task DeleteAsync(Guid id, Guid companyId)
    {
        _logger.LogInformation("Deleting customer with id: {Id} for company: {CompanyId}", id, companyId);

        var customer = await _context.Customers
            .FirstOrDefaultAsync(c => c.Id == id);

        if (customer == null)
        {
            throw new ArgumentException("Cliente não encontrado");
        }

        // Validate customer belongs to company
        if (customer.CompanyId != companyId)
        {
            throw new ArgumentException("Cliente não pertence a esta empresa");
        }

        _context.Customers.Remove(customer);
        await _context.SaveChangesAsync();

        _logger.LogInformation("Customer deleted successfully");
    }

    // Company-scoped methods
    public async Task<List<CustomerDto>> FindAllByCompanyAsync(Guid companyId)
    {
        _logger.LogDebug("Finding all customers for company: {CompanyId}", companyId);

        var customers = await _context.Customers
            .Where(c => c.CompanyId == companyId)
            .OrderBy(c => c.Name ?? c.Phone)
            .ToListAsync();

        return customers.Select(ToDto).ToList();
    }

    public async Task<CustomerDto?> FindByPhoneAndCompanyAsync(string phone, Guid companyId)
    {
        _logger.LogDebug("Finding customer by phone: {Phone} for company: {CompanyId}", phone, companyId);

        var customer = await _context.Customers
            .FirstOrDefaultAsync(c => c.Phone == phone && c.CompanyId == companyId);

        return customer != null ? ToDto(customer) : null;
    }

    public async Task<List<CustomerDto>> FindActiveByCompanyAsync(Guid companyId)
    {
        _logger.LogDebug("Finding active customers for company: {CompanyId}", companyId);

        var customers = await _context.Customers
            .Where(c => !c.IsBlocked && c.CompanyId == companyId)
            .OrderBy(c => c.Name ?? c.Phone)
            .ToListAsync();

        return customers.Select(ToDto).ToList();
    }

    public async Task<List<CustomerDto>> SearchByNameOrPhoneAndCompanyAsync(string searchTerm, Guid companyId)
    {
        _logger.LogDebug("Searching customers by term: {SearchTerm} for company: {CompanyId}", searchTerm, companyId);

        var customers = await _context.Customers
            .Where(c => c.CompanyId == companyId && 
                       (c.Name != null && c.Name.Contains(searchTerm) || 
                        c.Phone.Contains(searchTerm)))
            .OrderBy(c => c.Name ?? c.Phone)
            .ToListAsync();

        return customers.Select(ToDto).ToList();
    }

    public async Task<long> CountActiveByCompanyAsync(Guid companyId)
    {
        return await _context.Customers
            .Where(c => !c.IsBlocked && c.CompanyId == companyId)
            .CountAsync();
    }

    public async Task DeleteAllByCompanyAsync(Guid companyId)
    {
        _logger.LogInformation("Deleting all customers for company: {CompanyId}", companyId);

        var customers = await _context.Customers
            .Where(c => c.CompanyId == companyId)
            .ToListAsync();

        _context.Customers.RemoveRange(customers);
        await _context.SaveChangesAsync();

        _logger.LogInformation("Deleted {Count} customers for company: {CompanyId}", customers.Count, companyId);
    }

    private static CustomerDto ToDto(Customer customer)
    {
        return new CustomerDto
        {
            Id = customer.Id,
            CompanyId = customer.CompanyId,
            Phone = customer.Phone,
            Name = customer.Name,
            WhatsappId = customer.WhatsappId,
            ProfileUrl = customer.ProfileUrl,
            IsBlocked = customer.IsBlocked,
            SourceSystemName = customer.SourceSystemName,
            SourceSystemId = customer.SourceSystemId,
            ImportedAt = customer.ImportedAt,
            BirthDate = customer.BirthDate,
            LastDonationDate = customer.LastDonationDate,
            NextEligibleDonationDate = customer.NextEligibleDonationDate,
            BloodType = customer.BloodType,
            Height = customer.Height,
            Weight = customer.Weight,
            AddressStreet = customer.AddressStreet,
            AddressNumber = customer.AddressNumber,
            AddressComplement = customer.AddressComplement,
            AddressPostalCode = customer.AddressPostalCode,
            AddressCity = customer.AddressCity,
            AddressState = customer.AddressState,
            Email = customer.Email,
            Cpf = customer.Cpf,
            Rg = customer.Rg,
            RhFactor = customer.RhFactor,
            CreatedAt = customer.CreatedAt,
            UpdatedAt = customer.UpdatedAt
        };
    }
}